source/_posts/2015-04-13-working-with-init.markdown

```diff
@@ -0,0 +1,104 @@
+---
+layout: post
+title: "working with init"
+date: 2015-04-13 10:05:01 -0700
+comments: true
+categories:
+published: false 
+---
+##Create a linux startup service and init.d script from scratch, using chef and debian linux. 
+
+By C. Thomas Clark https://github.com/cthomaspdx
+
+As a Devops engineer at Househappy.org, I have been tasked to help maintain up-time for our servers and services they provide. If a server does have to reboot unexpectedly it would be nice to have some kind of assurance that when our servers come back online they automatically start their services. 
+
+Case in point: Our cloud server provider linode, very subtle notified us that our servers were due for an update and needed to be rebooted inorder for the update to take effect. Of course they formatted the email to look almost identical to a lot of the other email noise they send us, and I failed to recognize that this was serious and need my urgent attention. So time passed and the weekend rolled around, and all of our pager services started notifing us of our impending doom. One of the major problem with this is that many of our homegrown services with out init.d script failed to start back up after reboot. I would like to not have to worry about that happening again, so I decided to do my research. 
```

Comment:
Consider leaving out the actual story about how our provider rebooted our servers when we weren't expecting it and we didn't have init.d scripts to restart the services. Replace it with a hypothetical story about how something like that could happen. Make up something like: "What if the provider performs an emergency reboot of your server a 2 AM in the morning?" It reads more professional that way.